src/overlays/actors/ovl_En_Sb/z_en_sb.c

```diff
@@ -21,44 +36,373 @@ const ActorInit En_Sb_InitVars = {
+}
+
+void EnSb_SpawnBubbles(GlobalContext* globalCtx, EnSb* this) {
+    s32 bubbleCount;
+
+    if (this->actor.yDistToWater > 0.0f) {
+        for (bubbleCount = 0; bubbleCount < 10; bubbleCount++) {
+            EffectSsBubble_Spawn(globalCtx, &this->actor.world.pos, 10.0f, 10.0f, 30.0f, 0.25f);
+        }
+    }
+}
+
+void EnSb_SetupWaitClosed(EnSb* this) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_0600004C, 1.0f, 0, SkelAnime_GetFrameCount(&D_0600004C.common), 2, 0.0f);
+    this->state = SHELLBLADE_WAIT_CLOSED;
+    this->actionFunc = EnSb_Idle;
+}
+
+void EnSb_SetupOpen(EnSb* this) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06000194, 1.0f, 0, SkelAnime_GetFrameCount(&D_06000194.common), 2, 0.0f);
+    this->state = SHELLBLADE_OPEN;
+    this->actionFunc = EnSb_Open;
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_KUSAMUSHI_VIBE);
+}
+
+void EnSb_SetupWaitOpen(EnSb* this) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06002C8C, 1.0f, 0, SkelAnime_GetFrameCount(&D_06002C8C.common), 0, 0.0f);
+    this->state = SHELLBLADE_WAIT_OPEN;
+    this->actionFunc = EnSb_WaitOpen;
+}
+
+void EnSb_SetupLunge(EnSb* this) {
+    f32 frameCount = SkelAnime_GetFrameCount(&D_06000124.common);
+    f32 playbackSpeed = this->actor.yDistToWater > 0.0f ? 1.0f : 0.0f;
+
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06000124, playbackSpeed, 0.0f, frameCount, 2, 0);
+    this->state = SHELLBLADE_LUNGE;
+    this->actionFunc = EnSb_Lunge;
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_KUSAMUSHI_VIBE);
+}
+
+void EnSb_SetupBounce(EnSb* this) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_060000B4, 1.0f, 0, SkelAnime_GetFrameCount(&D_060000B4.common), 2, 0.0f);
+    this->state = SHELLBLADE_BOUNCE;
+    this->actionFunc = EnSb_Bounce;
+}
+
+void EnSb_SetupIdle(EnSb* this, s32 changeSpeed) {
+    f32 frameCount = SkelAnime_GetFrameCount(&D_0600004C.common);
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Sb_0x808E3EF0/EnSb_Destroy.asm")
+    if (this->state != SHELLBLADE_WAIT_CLOSED) {
+        SkelAnime_ChangeAnim(&this->skelAnime, &D_0600004C, 1.0f, 0, frameCount, 2, 0.0f);
+    }
+    this->state = SHELLBLADE_WAIT_CLOSED;
+    if (changeSpeed) {
+        if (this->actor.yDistToWater > 0.0f) {
+            this->actor.speedXZ = -5.0f;
+            if (this->actor.velocity.y < 0.0f) {
+                this->actor.velocity.y = 2.1f;
+            }
+        } else {
+            this->actor.speedXZ = -6.0f;
+            if (this->actor.velocity.y < 0.0f) {
+                this->actor.velocity.y = 1.4f;
+            }
+        }
+    }
+    this->attackTimer = 60;
+    this->actionFunc = EnSb_ReturnToIdle;
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Sb_0x808E3EF0/func_808E401C.asm")
+void EnSb_Idle(EnSb* this, GlobalContext* globalCtx) {
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 0xA, 0x7D0, 0);
+    if (this->actor.xzDistToPlayer <= 240.0f && this->actor.xzDistToPlayer > 0.0f) {
```

Comment:
```suggestion
    if ((this->actor.xzDistToPlayer <= 240.0f) && (this->actor.xzDistToPlayer > 0.0f)) {
```
More of a styling choice